### ISSUE TITLE
False alarm for superfluous-parens

### ISSUE THREAD

**Issue by kayhayen** [PyCQA/pylint#5803]
### Bug description

Simple code gives false warning:

```    
def f():
   return (base or dist_name) in ("Debian", "Ubuntu") 
```

As you can see, there are no braces around the return value, but I still get `superfluous-parens   Unnecessary parens after 'return' keyword`

### Configuration

```ini
Default config too.
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
nuitka/utils/Utils.py:107 C0325 superfluous-parens  Unnecessary parens after 'return' keyword
```

### Expected behavior

Should detect the parens usage properly.

### Pylint version

```shell
pylint 2.12.1
astroid 2.9.0
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
```

### OS / Environment

Debian Bullseye, but pip installed pylint

### Additional dependencies

Nothing
